### DISCUSSION TITLE
MAC-Adresse

### DISCUSSION THREAD

**Discussion by elmarfaber** [arendst/Tasmota#13898] [Q&A] [answered]
Hallo,

habe gerade 4 SONOFF Geräte 2x SONOFF Pow und 2x SONOFF Basic geflashed. Nun habe ich festgestellt, dass alle Geräte die gleiche MAC-Adresse haben und somit im Netzwerk alle die gleiche IP Adresse zugewiesen bekommen.
Habe ich fehlerhafte Geräte beim Chinesen gekauft??? 
Kann ich die Hardware Adresse beim selbst kompilieren eventuell verändern?

Viele Grüße

Elmar

**Comment by Jason2866** [comment, marked as answer]
Please post your questions in english. You can use https://www.deepl.com/de/translator for translating
> Hello,
>
> I have just flashed 4 SONOFF devices 2x SONOFF Pow and 2x SONOFF Basic. Now I have noticed that all
> devices have the same MAC address and are therefore all assigned the same IP address in the network.
> Did I buy faulty devices from a Chinese dealer?
> Can I possibly change the hardware address when compiling myself?
>
> Many greetings
>
> Elmar

I doubt that the devices do have the same MAC address since it is factory "burned" in the esp8266 chip and can not be changed. You have a network issue "somewhere"

**Comment by elmarfaber** [comment]
Thank you for your answer but I can exclude a network problem. The hostname (tasmota-23E867-2151) and MAC address (5C:CF:7F:23:E8:67) are always the same (5C-CF-7F = Espressif Inc.) and 23E867 (23-E8-67) the last digits are also always the same. The firmware generates the hostname via the MAC address. I bought all the devices last year via Aliexpress.

**Reply by s-hadinger** [reply]
I've never heard such thing. There's nothing we can do to help you. 

Are they genuine Sonoff?

**Reply by ascillato** [reply]
Have you tried to restart your router? Seems to be a router's issue.

**Reply by jevylux** [reply]
Hi, i have exactly the same problem. I use ESP8266 Relay modules as well as some sonoff switches. All of them have MAC  E8:DB:84:C5:FC:06. In the Arduino IDE it is possible to force a new mac address with the line
esp_wifi_set_mac(WIFI_IF_STA, &newMACAddress[0]);
Would it be possible to add such a function in tasmota, and in the interface of tasmotizer , to allow the change of the mac address ?
I exclude a network problem, as the Mac Addresses are displayed when tasmotizing via the serial board so no ethernet involved yet.

**Reply by Jason2866** [reply]
> Hi, i have exactly the same problem. I use ESP8266 Relay modules as well as some sonoff switches. All of them have MAC E8:DB:84:C5:FC:06. In the Arduino IDE it is possible to force a new mac address with the line esp_wifi_set_mac(WIFI_IF_STA, &newMACAddress[0]); Would it be possible to add such a function in tasmota, and in the interface of tasmotizer , to allow the change of the mac address ? I exclude a network problem, as the Mac Addresses are displayed when tasmotizing via the serial board so no ethernet involved yet.

I never had two esp chips with the same MAC address. There is something "fishy" in you network

**Reply by Jason2866** [reply]
If the have the same MAC address the are fake esp8266. Do not trust. We will not support fake chips
